### ISSUE TITLE
Setting `overflow-y: visible` doesn't affect the the value in the OM

### ISSUE THREAD

**Issue by jyc** [servo/servo#17555]
In the following testcase,

```html
<style>
input[type=text] {
  overflow-y: visible;
}
</style>

<div>
Search:
<input id="mytext" type="text">
Some more text
</div>

<script>
function print(prop) {
  console.log(prop + ": " + window.getComputedStyle(document.querySelector("input[type=text]")).getPropertyValue(prop));
}

print("overflow-y");
</script>
```

`visible` is output in Firefox, Chrome, and Safari, but Servo outputs `auto`.

**Comment by o0Ignition0o** [comment]
Hey @jdm , I would like to try to tackle this, if someone could mentor me :) 
I don't know yet how I'm gonna handle this though.
@highfive assign me

**Comment by jyc** [comment]
Hi @o0Ignition0o -- I'm very sorry, was just talking with @dholbert and it turns out I was wrong about the expected behavior, so I don't think this is actually a bug as-is.

So the UA style for `input` elements sets `overflow: hidden` (https://github.com/servo/servo/blob/master/resources/servo.css#L129). This isn't what the other browsers I tested do in their user-agent styles, which explains why they behave differently: the computed value of `overflow-x` and `overflow-y` is "as specified, except with visible computing to auto if one of overflow-x or overflow-y is not visible" (https://drafts.csswg.org/css-overflow-3/#propdef-overflow-x).

The `overflow: hidden` shorthand property declaration in the UA style causes `overflow-x` to be not `visible` even though we set `overflow-y: visible` in the test case, so the visible style computes to `auto`.

So I think everyone is behaving correctly, it's just Servo's unique UA style--I just messed up because I didn't look at the `overflow-y` property more carefully.
